Cython/Compiler/Parsing.py

```diff
@@ -3390,13 +3390,10 @@ def p_c_func_or_var_declaration(s, pos, ctx):
     else:
         is_const_method = 0
     if s.sy == '->':
-        # Special enough to give a better error message and keep going.
-        s.error(
```

Comment:
We should keep the error in the case that a type was already provided right after `c(p)def`.